src/tests/ftest/performance/performance_test_base.py

```diff
@@ -0,0 +1,407 @@
+        """A unique id for each test case ran."""
+        return "{}-{}".format(self.job_id, os.getpid())
+
+    def phase_barrier(self):
+        """Sleep barrier meant to be used between IO phases.
+
+        Useful for flushing system IO.
+        """
+        if self.phase_barrier_s > 0:
+            self.log.info("Sleeping for %s seconds", str(self.phase_barrier_s))
+            time.sleep(self.phase_barrier_s)
+
+    def _log_performance_params(self, group, extra_params=None):
+        """Log performance parameters.
+
+        Args:
+            group (str): ior or mdtest
+            extra_params (list, optional): extra params to print. Default is None.
+
+        """
+        group = group.upper()
+        if group not in ("IOR", "MDTEST"):
+            self.fail("Invalid group: {}".format(group))
+
+        # Start with common parameters
+        # Build a list of [PARAM_NAME, PARAM_VALUE]
+        params = [
+            ["TEST_ID", self.unique_id],
+            ["TEST_NAME", self.test_id],
+            ["TEST_GROUP", group],
+            ["NUM_SERVERS", self.num_servers],
+            ["NUM_TARGETS", self.num_targets],
+            ["NUM_CLIENTS", self.num_clients],
+            ["PPC", self.ppn],
+            ["PPN", self.ppn]
+        ]
+
+        # Get group-specific parameters
+        if group == "IOR":
+            params += [
+                ["API", self.ior_cmd.api.value],
+                ["OCLASS", self.ior_cmd.dfs_oclass.value],
+                ["XFER_SIZE", self.ior_cmd.transfer_size.value],
+                ["BLOCK_SIZE", self.ior_cmd.block_size.value],
+                ["SW_TIME", self.ior_cmd.sw_deadline.value],
+                ["CHUNK_SIZE", self.ior_cmd.dfs_chunk.value],
+                ["SEGMENTS", self.ior_cmd.segment_count.value],
+                ["ITERATIONS", self.ior_cmd.repetitions.value]
+            ]
+        elif group == "MDTEST":
+            params += [
+                ["API", self.mdtest_cmd.api.value],
+                ["OCLASS", self.mdtest_cmd.dfs_oclass.value],
+                ["DIR_OCLASS", self.mdtest_cmd.dfs_dir_oclass.value],
+                ["SW_TIME", self.mdtest_cmd.stonewall_timer.value],
+                ["CHUNK_SIZE", self.mdtest_cmd.dfs_chunk.value],
+                ["N_FILE", self.mdtest_cmd.num_of_files_dirs.value],
+                ["BYTES_READ", self.mdtest_cmd.read_bytes.value],
+                ["BYTES_WRITE", self.mdtest_cmd.write_bytes.value],
+                ["MDTEST_FLAGS", self.mdtest_cmd.flags.value]
+            ]
+
+        # Add the extra params
+        if extra_params is not None:
+            params += extra_params
+
+        # Print and align all parameters in the format:
+        # PARAM_NAME : PARAM_VALUE
+        max_len = max([len(param[0]) for param in params])
+        self.log_performance(
+            ["{:<{}} : {}".format(param[0], max_len, param[1]) for param in params]
+        )
+
+    def set_processes_ppn(self, namespace):
```

Comment:
I want to move this to a core file, but it requires some other changes. I'd like to do it as an improvement